src/datasets/arrow_dataset.py

```diff
@@ -2654,6 +2654,26 @@ def to_pandas(
                 for offset in range(0, len(self), batch_size)
             )
 
+    def add_column(self, name: str, column: Union[list, np.array]):
```

Comment:
```suggestion
    @transmit_format
    @fingerprint_transform(inplace=False)
    def add_column(self, name: str, column: Union[list, np.array], new_fingerprint: str):
```

The format of the dataset should be transmitted to the new dataset.
and the fingerprint updated accordingly